STYLE_GUIDE.md

```diff
@@ -0,0 +1,461 @@
+
+  ```pony
+  // OK
+  type foo is {(USize, USize): USize} box
+
+  // Not OK
+  type foo is {(USize, USize): USize } box
+
+  // Not OK
+  type foo is { (USize, USize): USize } box
+  ```
+
+### Type Aliases
+
+Align long union and intersection types with each type on a separate line starting with the `(`, `|`, or `&` symbols. In this case, every symbol must be separated by a single space or a newline.
+
+```pony
+// OK
+type Signed is (I8 | I16 | I32 | I64 | I128)
+
+// OK
+type Signed is
+  ( I8
+  | I16
+  | I32
+  | I64
+  | I128 )
+
+// OK
+type Signed is
+  ( I8
+  | I16
+  | I32
+  | I64
+  | I128
+  )
+
+// Not OK
+type Signed is ( I8
+              | I16
+              | I32
+              | I64
+              | I128 )
+
+// Not OK
+type Signed is
+  (I8
+  |I16
+  |I32
+  |I64
+  |I128)
+```
+
+### Array Literals
+
+Multiline arrays generally have a newline between each entry, though logical exceptions exist (such as tables).
+
+```pony
+// OK
+let ns = [as USize: 1; 2; 3]
+
+// OK
+let strs = ["a"; "b"; "c"]
+
+// OK
+let ns =
+  [ as USize:
+    1
+    2
+    3 ]
+
+// OK
+let ns =
+  [ as USize:
+    1
+    2
+    3
+  ]
+
+// Not OK
+let ns =
+  [as USize:
+   1
+   2
+   3
+  ]
+
+// Not OK
+let ns = [ as USize:
+    1
+    2
+    3
+]
+
+// Not OK
+let ns = [ as USize:
+          1
+          2
+          3 ]
+```
+
+### Control Structures
+
+- Control structures are aligned so that their keywords (if, else, for, try, end, etc.) are at the same indentation level.
+
+  ```pony
+  if cond1 then e1
+  elif cond2 then e2
+  else e3
+  end
+
+  if cond1 then
+    e1
+  elif cond2 then
+    e2
+  else
+    e3
+  end
+
+  for name in ["Bob"; "Fred"; "Sarah"].values() do
+    env.out.print(name)
+  end
+  ```
+
+- Small expressions may be placed on a single line.
+
+  ```pony
+  if cond then e1 else e2 end
+  ```
+
+- Match expression cases are __not__ indented.
+
+  ```pony
+  match (x, y)
+  | (None, _) => "none"
+  | (let s: String, 2) => s + " two"
+  | (let s: String, 3) => s + " three"
+  | (let s: String, let u: U32) if u > 14 => s + " other big integer"
+  | (let s: String, _) => s + " other small integer"
+  else "something else"
+  end
+
+  match alive
+  | true =>
+    Debug.err("Beep boop!")
+    this.forever()
+  | false =>
+    Debug.err("Ugah!")
+    None
+  end
+  ```
+
+- Match expressions __may not__ be on a single line.
+
+  ```pony
+  // OK
+  match x
+  | let n: U8 => n.string()
+  else "0"
+  end
+
+  // Not OK
+  match x | let n: U8 => n.string() else "0" end
+  ```
+
+### Assignment
+
+Assignment of a multiline expression is indented below the `=` symbol.
+
+```pony
+// OK
+var x =
+  if friendly then
+    "Hello"
+  else
+    false
+  end
+
+// OK
+let output =
+  recover String(
+    file_name.size()
+      + file_linenum.size()
+      + file_linepos.size()
+      + msg.size()
+      + 4)
+  end
+
+// Not OK
+var x = if friendly then
+  "Hello"
+else
+  false
+end
+```
+
+### Function Declarations
```

Comment:
Maybe say "method" instead of "function"? We use method to mean new/fun/be, generally. Definition vs declaration is a distinction in many languages (not just C/C++), but not in Pony.